diffrax/_integrate.py

```diff
@@ -227,14 +238,45 @@ def _maybe_static(static_x: Optional[ArrayLike], x: ArrayLike) -> ArrayLike:
         return x
 
 
+def _is_cond_fn(x: Any) -> bool:
+    return isinstance(x, Callable)
+
+
+def _compare_events(
+    old_event_result: Union[BoolScalarLike, RealScalarLike],
+    new_event_result: Union[BoolScalarLike, RealScalarLike],
+) -> BoolScalarLike:
+    return jnp.sign(jnp.array(old_event_result, float)) != jnp.sign(
+        jnp.array(new_event_result, float)
```

Comment:
Don't use the Python builtin `float` (or `int` etc.) as a dtype. This will actually give different behaviour on different platforms (MacOS vs Windows etc.) for some strange reason.
I think in this case you're probably looking for `dtype=jnp.result_type(old_event_result.dtype, jnp.float32)`?